python-sdk/nuscenes/scripts/export_kitti.py

```diff
@@ -0,0 +1,347 @@
+                print('Writing file: %s' % label_path)
+            with open(label_path, "w") as label_file:
+                for sample_annotation_token in sample_annotation_tokens:
+                    sample_annotation = self.nusc.get('sample_annotation', sample_annotation_token)
+
+                    # Get box in LIDAR frame.
+                    _, box_lidar_nusc, _ = self.nusc.get_sample_data(lidar_token, box_vis_level=BoxVisibility.NONE,
+                                                                     selected_anntokens=[sample_annotation_token])
+                    box_lidar_nusc = box_lidar_nusc[0]
+
+                    # Truncated: Set all objects to 0 which means untruncated.
+                    truncated = 0.0
+
+                    # Occluded: Hard-coded: Full visibility.
+                    occluded = 0
+
+                    # Convert nuScenes category to nuScenes detection challenge category.
+                    detection_name = category_to_detection_name(sample_annotation['category_name'])
+
+                    # Skip categories that are not part of the nuScenes detection challenge.
+                    if detection_name is None:
+                        continue
+
+                    # Convert to KITTI 3d and 2d box and KITTI output format.
+                    box_cam_kitti = KittiDB.box_nuscenes_to_kitti(
+                        box_lidar_nusc, Quaternion(matrix=velo_to_cam_rot), velo_to_cam_trans, r0_rect)
+                    box_cam_kitti.score = 0  # Set dummy score so we can use this file as result.
+                    output = KittiDB.box_to_string(name=detection_name, box=box_cam_kitti, truncation=truncated,
+                                                   occlusion=occluded)
+
+                    # Write to disk.
+                    label_file.write(output + '\n')
+
+    def render_kitti(self) -> None:
+        """
+        Renders the annotations in the KITTI dataset from a lidar and a camera view.
+        """
+        # Load the KITTI dataset.
+        kitti = KittiDB(root=self.nusc_kitti_dir, splits=(self.split,))
+
+        # Create output folder.
+        render_dir = os.path.join(self.nusc_kitti_dir, 'render')
+        if not os.path.isdir(render_dir):
+            os.mkdir(render_dir)
+
+        # Render each image.
+        for token in kitti.tokens[:self.image_count]:
+            for sensor in ['lidar', 'camera']:
+                out_path = os.path.join(render_dir, '%s_%s.png' % (token, sensor))
+                print('Rendering file to disk: %s' % out_path)
+                kitti.render_sample_data(token, sensor_modality=sensor, out_path=out_path)
+                plt.close()  # Close the windows to avoid a warning of too many open windows.
+
+    def kitti_res_to_nuscenes(self) -> None:
+        """
+        Converts a KITTI detection result to the nuScenes detection results format.
+        """
+        # Dummy meta data, please adjust accordingly.
+        meta = {
```

Comment:
am a bit unclear on what this meta data is

Reply:
Added meta as argument.